tests/test_console.py

```diff
@@ -122,6 +123,48 @@ def test_size():
     assert w == 99 and h == 101
 
 
+@pytest.mark.parametrize(
+    "is_windows,no_descriptor_size,stdin_size,stdout_size,stderr_size,expected_size",
+    [
+        # on Windows we'll use `os.get_terminal_size()` without arguments...
+        (True, (133, 24), ValueError, ValueError, ValueError, (133, 24)),
+        (False, (133, 24), ValueError, ValueError, ValueError, (80, 25)),
+        # ...while on other OS we'll try to pass stdin, then stdout, then stderr to it:
+        (False, ValueError, (133, 24), ValueError, ValueError, (133, 24)),
+        (False, ValueError, ValueError, (133, 24), ValueError, (133, 24)),
+        (False, ValueError, ValueError, ValueError, (133, 24), (133, 24)),
+        (False, ValueError, ValueError, ValueError, ValueError, (80, 25)),
+    ],
+)
+@mock.patch("rich.console.os.get_terminal_size")
+def test_size_can_fall_back_to_std_descriptors(
+    get_terminal_size_mock: mock.MagicMock,
+    is_windows: bool,
+    no_descriptor_size: tuple[int, int] | Type[ValueError],
+    stdin_size: tuple[int, int] | Type[ValueError],
+    stdout_size: tuple[int, int] | Type[ValueError],
+    stderr_size: tuple[int, int] | Type[ValueError],
+    expected_size,
+):
+    def get_terminal_size_mock_impl(fileno: int | None = None) -> tuple | ValueError:
+        value = {
+            None: no_descriptor_size,
+            sys.__stdin__.fileno(): stdin_size,
+            sys.__stdout__.fileno(): stdout_size,
+            sys.__stderr__.fileno(): stderr_size,
+        }[fileno]
+        if value is ValueError:
+            raise value
+        return value
+
+    get_terminal_size_mock.side_effect = get_terminal_size_mock_impl
+
+    console = Console(legacy_windows=False)
```

Comment:
looks like we always remove one pixel to the width if this is not forced to `False` ?

Reply:
Yeah, legacy windows treats new lines differently, they actually take up a cell in the terminal. We need to subtract 1 so the new line character doesn't go on to the following line.